### PR TITLE
feat(tidb-troubleshooting-map): 更新 TiKV 问题中 server is busy 指引说明

### PR DESCRIPTION
**Pull request by hzc989**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

update `tide-trouble-shooting-map.md`, update the tikv `server is busy` part

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

related Change Request https://github.com/pingcap/docs-cn/issues/8135
I'm participating the 2024-tidb-doc-dash https://github.com/orgs/pingcap/projects/51/views/3

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-troubleshooting-map.md [hunk 1]
```diff
@@ -218,6 +218,8 @@ TiDB 支持完整的分布式事务，自 v3.0 版本起，提供乐观事务与
 
 ### 4.3 客户端报 `server is busy` 错误
 
+自 TiDB 5.2 版本起，TiKV 引入新的流控机制代替之前的 RocksDB write stall 流控机制。该机制默认配置开启，TiKV 会自动关闭 KvDB 的 write stall 机制，还会关闭 RaftDB 中除 memtable 以外的 write stall 机制。因此客户端在默认情况下不会再收到诸如 `server is busy` 的错误信息，如你收到相关报错，请优先由于检查确认你的流控配置，并建议默认开启流控机制。关于流控机制的调优配置，参见[流控配置说明](https://docs.pingcap.com/zh/tidb/stable/tikv-configuration-file#storageflow-control)。如下为保留原有的客户端 `server is busy` 排错指引。
```
Comment (shawn0915) [inline tidb-troubleshooting-map.md:221]: 建议把 link 修改一下 `/tikv-configuration-file#storageflow-control`

File: tidb-troubleshooting-map.md [hunk 2]
```diff
@@ -218,6 +218,8 @@ TiDB 支持完整的分布式事务，自 v3.0 版本起，提供乐观事务与
 
 ### 4.3 客户端报 `server is busy` 错误
 
+自 TiDB 5.2 版本起，TiKV 引入新的流控机制代替之前的 RocksDB write stall 流控机制。该机制默认配置开启，TiKV 会自动关闭 KvDB 的 write stall 机制，还会关闭 RaftDB 中除 memtable 以外的 write stall 机制。因此客户端在默认情况下不会再收到诸如 `server is busy` 的错误信息，如你收到相关报错，请优先由于检查确认你的流控配置，并建议默认开启流控机制。关于流控机制的调优配置，参见[流控配置说明](/tikv-configuration-file#storageflow-control)。如下为保留原有的客户端 `server is busy` 排错指引。
```
Comment (Connor1996) [inline tidb-troubleshooting-map.md:221]: server is busy 有多种原因，还是会出现的。flow control 本身在流控也会导致 server is busy

**Comment by qiancai** [comment]
Hi @hzc989, I'm sorry to let you know that, according to the confirmation from @Connor1996 (a developer of TiKV), the issue [#8135](https://github.com/pingcap/docs-cn/issues/8135) is invalid and does not necessitate any documentation updates. Nonetheless, you'll still get the points for creating this PR in the Docs Dash 2024 event, and I sincerely appreciate your effort on resolving this issue.

If it is acceptable to you, could you please close this PR? Thanks.

**Review by Connor1996 — COMMENTED** [review]
可以根据这段文档更新下，还是有 ServerIsBusy 的情况对于 flow control
截屏2024-01-30 15 24 13

**Comment by qiancai** [comment]
@hzc989 PTAL

**Comment by hzc989** [comment]
> 可以根据这段文档更新下，还是有 ServerIsBusy 的情况对于 flow control 截屏2024-01-30 15 24 13

我仔细再去翻了下您当时的 work doc（https://github.com/tikv/tikv/issues/10137， https://github.com/tikv/rfcs/pull/67），大概明白你的意思了。重新写了一个版本，麻烦再卡卡看

File: tidb-troubleshooting-map.md [hunk 3]
```diff
@@ -218,6 +218,8 @@ TiDB 支持完整的分布式事务，自 v3.0 版本起，提供乐观事务与
 
 ### 4.3 客户端报 `server is busy` 错误
 
+首先，write stall 是一个 RocksDB 原生内建的性能降级机制。当 RocksDB 发生 write stall 时，整个系统的性能会出现剧烈下降。此前，TiDB 是通过直接给客户端返回 `ServerIsBusy` 错误来阻挡所有的写请求，带来剧烈的 QPS 性能下降。自 TiDB 5.2 版本起，TiKV 引入了新的流控机制，通过前置在调度层实现动态延迟写请求来达到抑制写入的目标，以期取代之前当遇到 write stall 时就给客户端返回 `server is busy` 来抑制写入的问题。该流控机制默认配置开启，TiKV 会自动关闭 KvDB 和 RaftDB (memtable 除外) 的 write stall 机制。但是，当 pending 的量超过一定阈值时，流控机制仍然会失效，开始拒绝部分或所有的写入请求，并返回客户端 `server is busy` 报错。相关阈值可参见[流控配置说明](/tikv-configuration-file#storageflow-control)。如下为保留原有的客户端 `server is busy` 排错指引。
```
Comment (Connor1996) [inline tidb-troubleshooting-map.md:221]: 这个应该放在 4.3.1 小节下

Comment (hzc989) [reply tidb-troubleshooting-map.md]: 再仔细研读了这一小节，的确应该放在 4.3.1 ，大佬麻烦再看看。

**Comment by qiancai** [comment]
@Connor1996 PTAL at the latest changes.  https://github.com/pingcap/docs-cn/pull/16193/files Thanks.

File: tidb-troubleshooting-map.md [hunk 4]
```diff
@@ -220,15 +220,13 @@ TiDB 支持完整的分布式事务，自 v3.0 版本起，提供乐观事务与
 
 通过查看监控：**Grafana** -> **TiKV** -> **errors** 确认具体 busy 原因。`server is busy` 是 TiKV 自身的流控机制，TiKV 通过这种方式告知 `tidb/ti-client` 当前 TiKV 的压力过大，稍后再尝试。
 
-- 4.3.1 TiKV RocksDB 出现 `write stall`。一个 TiKV 包含两个 RocksDB 实例，一个用于存储 Raft 日志，位于 `data/raft`。另一个用于存储真正的数据，位于 `data/db`。通过 `grep "Stalling" RocksDB` 日志查看 stall 的具体原因，RocksDB 日志是 LOG 开头的文件，LOG 为当前日志。
+- 4.3.1 TiKV RocksDB 出现 `write stall`。一个 TiKV 包含两个 RocksDB 实例，一个用于存储 Raft 日志，位于 `data/raft`。另一个用于存储真正的数据，位于 `data/db`。通过 `grep "Stalling" RocksDB` 日志查看 stall 的具体原因，RocksDB 日志是 LOG 开头的文件，LOG 为当前日志。`write stall` 是一个 RocksDB 原生内建的性能降级机制。当 RocksDB 发生 `write stall` 时，整个系统的性能会出现剧烈下降。此前，TiDB 是通过直接给客户端返回 `ServerIsBusy` 错误来阻挡所有的写请求，但这容易带来剧烈的 QPS 性能下降。自 5.2 版本起，TiKV 引入了新的流控机制，通过前置在调度层实现动态延迟写请求来达到抑制写入的目标，以期取代之前当遇到 `write stall` 时就给客户端返回 `server is busy` 来抑制写入的问题。新的流控机制默认配置开启，TiKV 会自动关闭 `KvDB` 和 `RaftDB` (memtable 除外) 的 `write stall` 机制。但是，当 pending 的量超过一定阈值时，流控机制仍然会失效，开始拒绝部分或所有的写入请求，并返回客户端 `server is busy` 报错，表现如下。更具体的说明和阈值可参考[流控配置说明](/tikv-configuration-file#storageflow-control)
```
Comment (Connor1996) [inline tidb-troubleshooting-map.md:223]:
```suggestion
- 4.3.1 TiKV RocksDB 出现 `write stall`。一个 TiKV 包含两个 RocksDB 实例，一个用于存储 Raft 日志，位于 `data/raft`。另一个用于存储真正的数据，位于 `data/db`。通过 `grep "Stalling" RocksDB` 日志查看 stall 的具体原因，RocksDB 日志是 LOG 开头的文件，LOG 为当前日志。`write stall` 是一个 RocksDB 原生内建的性能降级机制。当 RocksDB 发生 `write stall` 时，整个系统的性能会出现剧烈下降。此前，TiDB 是通过直接给客户端返回 `ServerIsBusy` 错误来阻挡所有的写请求，但这容易带来剧烈的 QPS 性能下降。自 5.2 版本起，TiKV 引入了新的流控机制，通过前置在调度层实现动态延迟写请求来达到抑制写入的目标，以期取代之前当遇到 `write stall` 时就给客户端返回 `server is busy` 来抑制写入的问题。新的流控机制默认配置开启，TiKV 会自动关闭 `KvDB` 和 `RaftDB` (memtable 除外) 的 `write stall` 机制。但是，当 pending 的量超过一定阈值时，流控机制仍然会生效，开始拒绝部分或所有的写入请求，并返回客户端 `server is busy` 报错，表现如下。更具体的说明和阈值可参考[流控配置说明](/tikv-configuration-file#storageflow-control)
```

**Review by Connor1996 — APPROVED** [review]
LGTM

File: tidb-troubleshooting-map.md [hunk 5]
```diff
@@ -220,15 +220,13 @@ TiDB 支持完整的分布式事务，自 v3.0 版本起，提供乐观事务与
 
 通过查看监控：**Grafana** -> **TiKV** -> **errors** 确认具体 busy 原因。`server is busy` 是 TiKV 自身的流控机制，TiKV 通过这种方式告知 `tidb/ti-client` 当前 TiKV 的压力过大，稍后再尝试。
 
-- 4.3.1 TiKV RocksDB 出现 `write stall`。一个 TiKV 包含两个 RocksDB 实例，一个用于存储 Raft 日志，位于 `data/raft`。另一个用于存储真正的数据，位于 `data/db`。通过 `grep "Stalling" RocksDB` 日志查看 stall 的具体原因，RocksDB 日志是 LOG 开头的文件，LOG 为当前日志。
+- 4.3.1 TiKV RocksDB 出现 `write stall`。一个 TiKV 包含两个 RocksDB 实例，一个用于存储 Raft 日志，位于 `data/raft`。另一个用于存储真正的数据，位于 `data/db`。通过 `grep "Stalling" RocksDB` 日志查看 stall 的具体原因，RocksDB 日志是 LOG 开头的文件，LOG 为当前日志。`write stall` 是一个 RocksDB 原生内建的性能降级机制。当 RocksDB 发生 `write stall` 时，整个系统的性能会出现剧烈下降。此前，TiDB 是通过直接给客户端返回 `ServerIsBusy` 错误来阻挡所有的写请求，但这容易带来剧烈的 QPS 性能下降。自 5.2 版本起，TiKV 引入了新的流控机制，通过前置在调度层实现动态延迟写请求来达到抑制写入的目标，以期取代之前当遇到 `write stall` 时就给客户端返回 `server is busy` 来抑制写入的问题。新的流控机制默认配置开启，TiKV 会自动关闭 `KvDB` 和 `RaftDB` (memtable 除外) 的 `write stall` 机制。但是，当 pending 的量超过一定阈值时，流控机制仍然会生效，开始拒绝部分或所有的写入请求，并返回客户端 `server is busy` 报错，表现如下。更具体的说明和阈值可参考[流控配置说明](/tikv-configuration-file#storageflow-control)
```
Comment (qiancai) [inline tidb-troubleshooting-map.md:223]:
```suggestion
- 4.3.1 TiKV RocksDB 出现 `write stall`。一个 TiKV 包含两个 RocksDB 实例，一个用于存储 Raft 日志，位于 `data/raft`。另一个用于存储真正的数据，位于 `data/db`。通过 `grep "Stalling" RocksDB` 日志查看 stall 的具体原因，RocksDB 日志是 LOG 开头的文件，LOG 为当前日志。`write stall` 是一个 RocksDB 原生内建的性能降级机制。当 RocksDB 发生 `write stall` 时，整个系统的性能会急剧下降。在 v5.2.0 之前，TiDB 通过直接给客户端返回 `ServerIsBusy` 错误来阻挡所有的写请求，但这容易导致 QPS 性能急剧下降。自 v5.2.0 起，TiKV 引入了新的流控机制，通过前置在调度层实现动态延迟写请求来抑制写入，以替代之前遇到 `write stall` 时就给客户端返回 `server is busy` 来抑制写入的机制。新的流控机制默认配置开启，TiKV 会自动关闭 `KvDB` 和 `RaftDB` (memtable 除外) 的 `write stall` 机制。但是，当 pending 的请求量超过一定阈值时，流控机制仍然会生效，开始拒绝部分或所有的写入请求，并返回客户端 `server is busy` 报错，表现如下。详细的说明和阈值可参考[流控配置说明](/tikv-configuration-file#storageflow-control)
```

File: tidb-troubleshooting-map.md [hunk 6]
```diff
@@ -220,15 +220,13 @@ TiDB 支持完整的分布式事务，自 v3.0 版本起，提供乐观事务与
...
 
-    - `level0 sst` 太多导致 stall，可以添加参数 `[rocksdb] max-sub-compactions = 2`（或者 `3`），加快 level0 sst 往下 compact 的速度。该参数的意思是将从 level0 到 level1 的 compaction 任务最多切成 `max-sub-compactions` 个子任务交给多线程并发执行，见案例 [case-815](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case815.md)。
+    - `pending compaction bytes` 太多导致触发新流控机制开始工作， 可以通过调大 `soft-pending-compaction-bytes-limit` 和 `hard-pending-compaction-bytes-limit` 参数来缓解：
```
Comment (qiancai) [inline tidb-troubleshooting-map.md:225]:
```suggestion
    - 如果 `pending compaction bytes` 太多触发新流控机制开始工作，可以通过调大 `soft-pending-compaction-bytes-limit` 和 `hard-pending-compaction-bytes-limit` 参数来缓解。
```

File: tidb-troubleshooting-map.md [hunk 7]
```diff
@@ -220,15 +220,13 @@ TiDB 支持完整的分布式事务，自 v3.0 版本起，提供乐观事务与
...
 
-    - `pending compaction bytes` 太多导致 stall，磁盘 I/O 能力在业务高峰跟不上写入，可以通过调大对应 Column Family (CF) 的 `soft-pending-compaction-bytes-limit` 和 `hard-pending-compaction-bytes-limit` 参数来缓解：
+        - 如果 `pending compaction bytes` 达到该阈值，流控就会开始拒绝一部分的写请求（通过给客户端返回`ServerIsBusy`）。默认值 192GB，`[storage.flow-control] soft-pending-compaction-bytes-limit = "384GB"`。
```
Comment (qiancai) [inline tidb-troubleshooting-map.md:227]:
```suggestion
        - 如果 `pending compaction bytes` 达到 `soft-pending-compaction-bytes-limit` 参数的值（默认为 `192GB`），流控就会开始拒绝一部分的写请求（通过给客户端返回`ServerIsBusy`）。此时，可以调大该参数的值，例如，`[storage.flow-control] soft-pending-compaction-bytes-limit = "384GB"`。
```

File: tidb-troubleshooting-map.md [hunk 8]
```diff
@@ -220,15 +220,13 @@ TiDB 支持完整的分布式事务，自 v3.0 版本起，提供乐观事务与
...
 
-        - 如果 `pending compaction bytes` 达到该阈值，RocksDB 会放慢写入速度。默认值 64GB，`[rocksdb.defaultcf] soft-pending-compaction-bytes-limit = "128GB"`。
-
-        - 如果 `pending compaction bytes` 达到该阈值，RocksDB 会 stop 写入，通常不太可能触发该情况，因为在达到 `soft-pending-compaction-bytes-limit` 的阈值之后会放慢写入速度。默认值 256GB，`hard-pending-compaction-bytes-limit = "512GB"`<!--见案例 [case-275](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case275.md) -->。
+        - 如果 `pending compaction bytes` 达到该阈值，流控就会开始拒绝所有的写请求（通过给客户端返回`ServerIsBusy`），通常不太可能触发该情况，因为在达到 `soft-pending-compaction-bytes-limit` 的阈值之后流控机制就会介入而放慢写入速度。默认值 1024GB，`[storage.flow-control] hard-pending-compaction-bytes-limit = "2048GB"`<!--见案例 [case-275](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case275.md) -->。
```
Comment (qiancai) [inline tidb-troubleshooting-map.md:229]:
```suggestion
        - 如果 `pending compaction bytes` 达到 `hard-pending-compaction-bytes-limit` 参数的值（默认为 1024GB），流控就会开始拒绝所有的写请求（通过给客户端返回`ServerIsBusy`）。通常不太可能触发该情况，因为在达到 `soft-pending-compaction-bytes-limit` 的阈值之后流控机制就会介入而放慢写入速度。如果触发，可以调大该参数的值，例如，`[storage.flow-control] hard-pending-compaction-bytes-limit = "2048GB"`<!--见案例 [case-275](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case275.md) -->。
```

**Comment by qiancai** [comment]
@hzc989 I've added some comments to this PR. PTAL. Thanks.

**Review by qiancai — APPROVED** [review]

**Comment by qiancai** [comment]
Hi @hzc989, this PR looks good to me now. Would you please update the English PR https://github.com/pingcap/docs/pull/16172 accordingly? Thanks.

File: tidb-troubleshooting-map.md [hunk 9]
```diff
@@ -220,15 +220,13 @@ TiDB 支持完整的分布式事务，自 v3.0 版本起，提供乐观事务与
...
 
-    - `level0 sst` 太多导致 stall，可以添加参数 `[rocksdb] max-sub-compactions = 2`（或者 `3`），加快 level0 sst 往下 compact 的速度。该参数的意思是将从 level0 到 level1 的 compaction 任务最多切成 `max-sub-compactions` 个子任务交给多线程并发执行，见案例 [case-815](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case815.md)。
+    - 如果 `pending compaction bytes` 太多触发 `server is busy` 报错，可以通过调大 `soft-pending-compaction-bytes-limit` 和 `hard-pending-compaction-bytes-limit` 参数来缓解。
 
-    - `pending compaction bytes` 太多导致 stall，磁盘 I/O 能力在业务高峰跟不上写入，可以通过调大对应 Column Family (CF) 的 `soft-pending-compaction-bytes-limit` 和 `hard-pending-compaction-bytes-limit` 参数来缓解：
+        - 如果 `pending compaction bytes` 达到 `soft-pending-compaction-bytes-limit` 参数的值（默认为 `192GB`），流控就会开始拒绝一部分的写请求（通过给客户端返回 `ServerIsBusy`）。此时，可以调大该参数的值，例如，`[storage.flow-control] soft-pending-compaction-bytes-limit = "384GB"`。
```
Comment (qiancai) [inline tidb-troubleshooting-map.md:227]:
```suggestion
        - 如果 `pending compaction bytes` 达到 `soft-pending-compaction-bytes-limit` 参数的值（默认为 `192GiB`），流控就会开始拒绝一部分的写请求（通过给客户端返回 `ServerIsBusy`）。此时，可以调大该参数的值，例如，`[storage.flow-control] soft-pending-compaction-bytes-limit = "384GiB"`。
```

File: tidb-troubleshooting-map.md [hunk 10]
```diff
@@ -220,15 +220,13 @@ TiDB 支持完整的分布式事务，自 v3.0 版本起，提供乐观事务与
...
 
-        - 如果 `pending compaction bytes` 达到该阈值，RocksDB 会放慢写入速度。默认值 64GB，`[rocksdb.defaultcf] soft-pending-compaction-bytes-limit = "128GB"`。
-
-        - 如果 `pending compaction bytes` 达到该阈值，RocksDB 会 stop 写入，通常不太可能触发该情况，因为在达到 `soft-pending-compaction-bytes-limit` 的阈值之后会放慢写入速度。默认值 256GB，`hard-pending-compaction-bytes-limit = "512GB"`<!--见案例 [case-275](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case275.md) -->。
+        - 如果 `pending compaction bytes` 达到 `hard-pending-compaction-bytes-limit` 参数的值（默认为 1024GB），流控就会开始拒绝所有的写请求（通过给客户端返回 `ServerIsBusy`）。通常不太可能触发该情况，因为在达到 `soft-pending-compaction-bytes-limit` 的阈值之后，流控机制就会介入而放慢写入速度。如果触发，可以调大该参数的值，例如，`[storage.flow-control] hard-pending-compaction-bytes-limit = "2048GB"`<!--见案例 [case-275](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case275.md) -->。
```
Comment (qiancai) [inline tidb-troubleshooting-map.md:229]:
```suggestion
        - 如果 `pending compaction bytes` 达到 `hard-pending-compaction-bytes-limit` 参数的值（默认为 1024GiB），流控就会开始拒绝所有的写请求（通过给客户端返回 `ServerIsBusy`）。通常不太可能触发该情况，因为在达到 `soft-pending-compaction-bytes-limit` 的阈值之后，流控机制就会介入而放慢写入速度。如果触发，可以调大该参数的值，例如，`[storage.flow-control] hard-pending-compaction-bytes-limit = "2048GiB"`<!--见案例 [case-275](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case275.md) -->。
```

File: tidb-troubleshooting-map.md [hunk 11]
```diff
@@ -220,15 +220,13 @@ TiDB 支持完整的分布式事务，自 v3.0 版本起，提供乐观事务与
...
+- 4.3.1 TiKV RocksDB 出现 `write stall`。一个 TiKV 包含两个 RocksDB 实例，一个用于存储 Raft 日志，位于 `data/raft`。另一个用于存储真正的数据，位于 `data/db`。通过 `grep "Stalling" RocksDB` 日志查看 stall 的具体原因，RocksDB 日志是 LOG 开头的文件，LOG 为当前日志。`write stall` 是一个 RocksDB 原生内建的性能降级机制。当 RocksDB 发生 `write stall` 时，系统的整体性能会急剧下降。在 v5.2.0 之前，TiDB 通过直接给客户端返回 `ServerIsBusy` 错误来阻挡所有的写请求，但这容易导致 QPS 性能急剧下降。自 v5.2.0 起，TiKV 引入了新的流控机制，通过前置在调度层实现动态延迟写请求来抑制写入，以替代之前遇到 `write stall` 时就给客户端返回 `server is busy` 来抑制写入的机制。新的流控机制默认配置开启，TiKV 会自动关闭 `KvDB` 和 `RaftDB` (memtable 除外) 的 `write stall` 机制。但是，当 pending 的请求量超过一定阈值时，流控机制仍然会生效，开始拒绝部分或所有的写入请求，并返回客户端 `server is busy` 报错，表现如下。详细的说明和阈值可参考[流控配置说明](/tikv-configuration-file#storageflow-control)
 
-    - `level0 sst` 太多导致 stall，可以添加参数 `[rocksdb] max-sub-compactions = 2`（或者 `3`），加快 level0 sst 往下 compact 的速度。该参数的意思是将从 level0 到 level1 的 compaction 任务最多切成 `max-sub-compactions` 个子任务交给多线程并发执行，见案例 [case-815](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case815.md)。
+    - 如果 `pending compaction bytes` 太多触发 `server is busy` 报错，可以通过调大 `soft-pending-compaction-bytes-limit` 和 `hard-pending-compaction-bytes-limit` 参数来缓解。
```
Comment (qiancai) [inline tidb-troubleshooting-map.md:225]:
```suggestion
    - 如果 pending compaction bytes 太多触发 `server is busy` 报错，可以通过调大 `soft-pending-compaction-bytes-limit` 和 `hard-pending-compaction-bytes-limit` 参数来缓解。
```

File: tidb-troubleshooting-map.md (hunk 10 above)
Comment (qiancai) [reply tidb-troubleshooting-map.md:229]: <https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case275.md> is no longer available.

File: tidb-troubleshooting-map.md [hunk 12]
```diff
@@ -220,15 +220,13 @@ TiDB 支持完整的分布式事务，自 v3.0 版本起，提供乐观事务与
 
 通过查看监控：**Grafana** -> **TiKV** -> **errors** 确认具体 busy 原因。`server is busy` 是 TiKV 自身的流控机制，TiKV 通过这种方式告知 `tidb/ti-client` 当前 TiKV 的压力过大，稍后再尝试。
 
-- 4.3.1 TiKV RocksDB 出现 `write stall`。一个 TiKV 包含两个 RocksDB 实例，一个用于存储 Raft 日志，位于 `data/raft`。另一个用于存储真正的数据，位于 `data/db`。通过 `grep "Stalling" RocksDB` 日志查看 stall 的具体原因，RocksDB 日志是 LOG 开头的文件，LOG 为当前日志。
+- 4.3.1 TiKV RocksDB 出现 `write stall`。一个 TiKV 包含两个 RocksDB 实例，一个用于存储 Raft 日志，位于 `data/raft`。另一个用于存储真正的数据，位于 `data/db`。通过 `grep "Stalling" RocksDB` 日志可以查看 stall 的具体原因，RocksDB 日志是 LOG 开头的文件，LOG 为当前日志。`write stall` 是一个 RocksDB 原生内建的性能降级机制。当 RocksDB 发生 `write stall` 时，系统的整体性能会急剧下降。在 v5.2.0 之前，当发生 `write stall` 时，TiDB 通过直接给客户端返回 `ServerIsBusy` 错误来阻挡所有的写请求，但这容易导致 QPS 性能急剧下降。自 v5.2.0 起，TiKV 引入了新的流控机制，通过前置在调度层实现动态延迟写请求来抑制写入，以替代之前遇到 `write stall` 时就给客户端返回 `server is busy` 来抑制写入的机制。新的流控机制默认配置开启，TiKV 会自动关闭 `KvDB` 和 `RaftDB` (memtable 除外) 的 `write stall` 机制。但是，当 pending 的请求量超过一定阈值时，流控机制仍然会生效，开始拒绝部分或所有的写入请求，并返回客户端 `server is busy` 报错，表现如下。详细的说明和阈值可参考[流控配置说明](/tikv-configuration-file#storageflow-control)。
```
Comment (hfxsd) [inline tidb-troubleshooting-map.md:223]:
```suggestion
- 4.3.1 TiKV RocksDB 出现 `write stall`。一个 TiKV 包含两个 RocksDB 实例，一个用于存储 Raft 日志，位于 `data/raft`。另一个用于存储真正的数据，位于 `data/db`。通过 `grep "Stalling" RocksDB` 日志可以查看 stall 的具体原因，RocksDB 日志是 `LOG` 开头的文件，`LOG` 为当前日志。`write stall` 是一个 RocksDB 原生内建的性能降级机制。当 RocksDB 发生 `write stall` 时，系统的整体性能会急剧下降。在 v5.2.0 之前，当发生 `write stall` 时，TiDB 通过直接给客户端返回 `ServerIsBusy` 错误来阻挡所有的写请求，但这容易导致 QPS 性能急剧下降。自 v5.2.0 起，TiKV 引入了新的流控机制，通过前置在调度层实现动态延迟写请求来抑制写入，以替代之前遇到 `write stall` 时就给客户端返回 `server is busy` 来抑制写入的机制。新的流控机制默认配置开启，TiKV 会自动关闭 `KvDB` 和 `RaftDB` (memtable 除外) 的 `write stall` 机制。但是，当 pending 的请求量超过一定阈值时，流控机制仍然会生效，开始拒绝部分或所有的写入请求，并返回客户端 `server is busy` 报错，表现如下。详细的说明和阈值可参考[流控配置说明](/tikv-configuration-file#storageflow-control)。
```

File: tidb-troubleshooting-map.md [hunk 13]
```diff
@@ -220,15 +220,13 @@ TiDB 支持完整的分布式事务，自 v3.0 版本起，提供乐观事务与
...
 
-    - `level0 sst` 太多导致 stall，可以添加参数 `[rocksdb] max-sub-compactions = 2`（或者 `3`），加快 level0 sst 往下 compact 的速度。该参数的意思是将从 level0 到 level1 的 compaction 任务最多切成 `max-sub-compactions` 个子任务交给多线程并发执行，见案例 [case-815](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case815.md)。
+    - 如果 pending compaction bytes 太多触发 `server is busy` 报错，可以通过调大 `soft-pending-compaction-bytes-limit` 和 `hard-pending-compaction-bytes-limit` 参数来缓解。
```
Comment (hfxsd) [inline tidb-troubleshooting-map.md:225]:
```suggestion
    - 如果 pending compaction bytes 太多触发 `server is busy` 报错，可以通过调大 `soft-pending-compaction-bytes-limit` 和 `hard-pending-compaction-bytes-limit` 参数的值来缓解。
```

File: tidb-troubleshooting-map.md [hunk 14]
```diff
@@ -220,15 +220,13 @@ TiDB 支持完整的分布式事务，自 v3.0 版本起，提供乐观事务与
...
 
-    - `pending compaction bytes` 太多导致 stall，磁盘 I/O 能力在业务高峰跟不上写入，可以通过调大对应 Column Family (CF) 的 `soft-pending-compaction-bytes-limit` 和 `hard-pending-compaction-bytes-limit` 参数来缓解：
+        - 如果 pending compaction bytes 达到 `soft-pending-compaction-bytes-limit` 参数的值（默认为 `192GiB`），流控就会开始拒绝一部分的写请求（通过给客户端返回 `ServerIsBusy`）。此时，可以调大该参数的值，例如，`[storage.flow-control] soft-pending-compaction-bytes-limit = "384GiB"`。
 
-        - 如果 `pending compaction bytes` 达到该阈值，RocksDB 会放慢写入速度。默认值 64GB，`[rocksdb.defaultcf] soft-pending-compaction-bytes-limit = "128GB"`。
-
-        - 如果 `pending compaction bytes` 达到该阈值，RocksDB 会 stop 写入，通常不太可能触发该情况，因为在达到 `soft-pending-compaction-bytes-limit` 的阈值之后会放慢写入速度。默认值 256GB，`hard-pending-compaction-bytes-limit = "512GB"`<!--见案例 [case-275](https://github.com/pingcap/tidb-map/blob/master/maps/diagnose-case-study/case275.md) -->。
+        - 如果 pending compaction bytes 达到 `hard-pending-compaction-bytes-limit` 参数的值（默认为 1024GiB），流控就会开始拒绝所有的写请求（通过给客户端返回 `ServerIsBusy`）。通常不太可能触发该情况，因为在达到 `soft-pending-compaction-bytes-limit` 的阈值之后，流控机制就会介入而放慢写入速度。如果触发，可以调大该参数的值，例如，`[storage.flow-control] hard-pending-compaction-bytes-limit = "2048GiB"`。
```
Comment (hfxsd) [inline tidb-troubleshooting-map.md:229]:
```suggestion
        - 如果 pending compaction bytes 达到 `hard-pending-compaction-bytes-limit` 参数的值（默认为 `"1024GiB"`），流控就会开始拒绝所有的写请求（通过给客户端返回 `ServerIsBusy`）。通常不太可能触发该情况，因为在达到 `soft-pending-compaction-bytes-limit` 的阈值之后，流控机制就会介入而放慢写入速度。如果触发，可以调大该参数的值，例如，`[storage.flow-control] hard-pending-compaction-bytes-limit = "2048GiB"`。
```

**Pull request merged**